legate/core/io.py

```diff
@@ -185,41 +199,42 @@ def ingest(
         raise TypeError(
             f"data_split: expected a DataSplit object but got {data_split}"
         )
-    if get_local_colors is None:
 
-        # Assign colors following the default sharding
-        def get_local_colors():
-            sid = _runtime.core_context.get_sharding_id(
-                _runtime.core_library.LEGATE_CORE_LINEARIZE_SHARD_ID
-            )
-            shard = legion.legion_runtime_local_shard(
-                _runtime.legion_runtime, _runtime.legion_context
-            )
-            domain = Rect(colors).raw()
-            total_shards = legion.legion_runtime_total_shards(
-                _runtime.legion_runtime, _runtime.legion_context
-            )
-            points_size = ffi.new("size_t *")
-            points_size[0] = 1
-            for c in colors:
-                points_size[0] *= c
-            points_ptr = ffi.new("legion_domain_point_t[%s]" % points_size[0])
-            legion.legion_sharding_functor_invert(
-                sid,
-                shard,
-                domain,
-                domain,
-                total_shards,
-                points_ptr,
-                points_size,
-            )
-            points = []
-            for i in range(points_size[0]):
-                points.append(Point(points_ptr[i]))
-            return points
+    # Assign colors following the default sharding
+    def default_get_local_colors() -> list[Point]:
+        sid = _runtime.core_context.get_sharding_id(
+            _runtime.core_library.LEGATE_CORE_LINEARIZE_SHARD_ID
+        )
+        shard = legion.legion_runtime_local_shard(
+            _runtime.legion_runtime, _runtime.legion_context
+        )
+        domain = Rect(colors).raw()
+        total_shards = legion.legion_runtime_total_shards(
+            _runtime.legion_runtime, _runtime.legion_context
+        )
+        points_size = ffi.new("size_t *")
+        points_size[0] = 1
+        for c in colors:
+            points_size[0] *= c
+        points_ptr = ffi.new("legion_domain_point_t[%s]" % points_size[0])
+        legion.legion_sharding_functor_invert(
+            sid,
+            shard,
+            domain,
+            domain,
+            total_shards,
+            points_ptr,
+            points_size,
+        )
+        points = []
```

Comment:
maybe it's outside the scope of this PR, but I'd prefer a for-comprehension here than an explicit loop calling `append`.

Reply:
I'm generally a fan of going small-scale code gardening as it is noticed https://github.com/nv-legate/legate.core/pull/196/commits/83b6367b1feadcd79dc2d163a3dc22e8dcd82322